cdap-ui/app/cdap/components/Overview/DatasetOverview/index.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import OverviewHeader from 'components/Overview/OverviewHeader';
+import OverviewMetaSection from 'components/Overview/OverviewMetaSection';
+import DatasetOverviewTab from 'components/Overview/DatasetOverview/DatasetOverviewTab';
+import NamespaceStore from 'services/NamespaceStore';
+import shortid from 'shortid';
+import {objectQuery} from 'services/helpers';
+import {MyDatasetApi} from 'api/dataset';
+import {MyMetadataApi} from 'api/metadata';
+import isNil from 'lodash/isNil';
+import T from 'i18n-react';
+
+
+export default class DatasetOverview extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      entity: this.props.entity,
+      entityDetail: null,
+      loading: false
+    };
+  }
+
+  componentWillMount() {
+    this.fetchDatasetDetail();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let {entity} = nextProps;
+    if (!isNil(entity)) {
+      this.setState({
+        entity,
+      }, this.fetchStreamDetail.bind(this));
+    }
+  }
+
+  fetchDatasetDetail() {
+    this.setState({
+      loading: true
+    });
+    let namespace = NamespaceStore.getState().selectedNamespace;
+    if (objectQuery(this.props, 'entity', 'id')) {
+      const datasetParams = {
+        namespace,
+        datasetId: this.props.entity.id
+      };
+
+      const metadataParams = {
+        namespace,
+        entityType: 'datasets',
+        entityId: this.props.entity.id,
+        scope: 'SYSTEM'
+      };
+
+      MyMetadataApi.getProperties(metadataParams)
+        .combineLatest(MyDatasetApi.getPrograms(datasetParams))
+        .subscribe((res) => {
+          let programs = res[1].map((program) => {
+            program.uniqueId = shortid.generate();
+            program.app = program.application.applicationId;
+            return program;
+          });
+
+          let entityDetail = {
```

Comment:
Yes, but whether the schema is empty or not should be handled by the SchemaTab I think